Documentation/ceph-csi-troubleshooting.md

```diff
@@ -441,3 +441,13 @@ $ rbd ls --id=csi-rbd-node -m=10.111.136.166:6789 --key=AQDpIQhg+v83EhAAgLboWIbl
 ```
 
 Where `-m` is one of the mon endpoints and the `--key` is the key used by the CSI driver for accessing the Ceph cluster.
+
+## Parallel RBD PVC creation hangs for new BlockPool
+
+This issue is specifically present in CephCSI `v3.4.x`, used by rook `>=v1.7.1` and occurs when multiple parallel PVCs creation requests are issued on a newly created blockpool. Follow the steps below to workaround the issue:
```

Comment:
this can happen for other reasons also. please add steps (provide rbd commands to run from the toolbox) to identify the issue.

Reply:
> this can happen for other reasons also. please add steps (provide rbd commands to run from the toolbox) to identify the issue.

I've linked the issue back in the docs. 

I don't think the issue can be identified by running commands from the toolbox(+ would add to the effort required).
 
The only direct method is seeing multiple reqs getting stuck at `creating rbd %s image using mons` log in csi-rbdplugin-provisioner but considering default csi log level is 0, this is not feasible. 

Directly implementing the steps would be easier, if it solves the issue then it's good otherwise something else would be at fault. IMO, its good to keep the number of steps short and easy as possible.

Reply:
> I don't think the issue can be identified by running commands from the toolbox(+ would add to the effort required).


Cant this be verified by creating the image in the toolbox?

> Directly implementing the steps would be easier, if it solves the issue then it's good otherwise something else would be at fault. IMO, its good to keep the number of steps short and easy as possible.


Why someone will blindly execute the command in a production cluster without knowing the issue?

Reply:
> > I don't think the issue can be identified by running commands from the toolbox(+ would add to the effort required).
> 
> Cant this be verified by creating the image in the toolbox?

creating the image in the toolbox will succeed but I don't think it will be enough to directly narrow it down to this issue.

> > Directly implementing the steps would be easier, if it solves the issue then it's good otherwise something else would be at fault. IMO, its good to keep the number of steps short and easy as possible.
> 
> Why someone will blindly execute the command in a production cluster without knowing the issue?

Again

> The only direct method is seeing multiple reqs getting stuck at creating rbd %s image using mons log in csi-rbdplugin-provisioner but considering default csi log level is 0, this is not feasible.

Changing csi log level leads to a restart, which may let a single create image request slip and succeed without deadlock, resolving the issue but may not help in identification of the issue.

Reply:
> creating the image in the toolbox will succeed but I don't think it will be enough to directly narrow it down to this issue.


 How this will succeed? its also using the same ceph library? am I missing something?

>Changing csi log level leads to a restart, which may let a single create image request slip and succeed without deadlock, resolving the issue but may not help in identification of the issue.


not suggesting this one for now.

Reply:
> > creating the image in the toolbox will succeed but I don't think it will be enough to directly narrow it down to this issue.
> 
> How this will succeed? its also using the same ceph library? am I missing something?


The deadlock happens when the same `rados connection` is used for creating image(which cephcsi does). 
The cmd from toolbox, creates a separate connection to the cluster and uses it(that's why it can be reproduced from cli).